osh/word_eval.py

```diff
@@ -2162,23 +2166,26 @@ def EvalRhsWord(self, UP_w):
             part0 = w.parts[0]
             UP_part0 = part0
             tag = part0.tag()
-            # Special case for a=(1 2).  ShArrayLiteral won't appear in words that
-            # don't look like assignments.
-            if tag == word_part_e.ShArrayLiteral:
-                part0 = cast(ShArrayLiteral, UP_part0)
-                array_words = part0.words
-                words = braces.BraceExpandWords(array_words)
-                strs = self.EvalWordSequence(words)
-                return bash_impl.BashArray_FromList(strs)
-
-            if tag == word_part_e.BashAssocLiteral:
-                part0 = cast(word_part.BashAssocLiteral, UP_part0)
-                d = NewDict()  # type: Dict[str, str]
+            if tag == word_part_e.InitializerLiteral:
+                part0 = cast(word_part.InitializerLiteral, UP_part0)
+
+                assigns = []  # type: List[InitializerValue]
                 for pair in part0.pairs:
-                    k = self.EvalWordToString(pair.key)
-                    v = self.EvalWordToString(pair.value)
-                    d[k.s] = v.s
-                return value.BashAssoc(d)
+                    UP_pair = pair
+                    if pair.tag() == InitializerWord_e.ArrayWord:
```

Comment:
It might be nicer to use the `with  tagswitch` idiom here , even though it causes another level of indentation

The old code could have used it too, although it did an early return

Reply:
850af58fe